hardware/blebox/CMakeLists.txt

```diff
@@ -0,0 +1,152 @@
+# BleBox devices integration
+
+# {{{ -------------------core library --------------------------------
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+
+# stuff common with domoticz and unit tests
+set( blebox_SRCS
+  db/annotate.cpp
+  db/device_status.cpp
+  db/hardware.cpp
+  db/migration.cpp
+  errors.cpp
+  box.cpp
+  rfx.cpp
+  version.cpp
+  products.cpp
+  widgets.cpp
+  widgets/widget.cpp
+  )
+
+add_library(blebox OBJECT ${blebox_SRCS})
+add_definitions(-DWEBSERVER_DONT_USE_ZIP)
+
+# for cereal
+target_include_directories(blebox PRIVATE ${CMAKE_CURRENT_LIST_DIR}/../..)
+
+# TODO: remove this workaround (for unzip.h header path)
+option(USE_BUILTIN_MINIZIP "Use builtin minizip library" YES)
+if(USE_BUILTIN_MINIZIP)
+  set(MINIZIP_INCLUDE_DIRS "${CMAKE_SOURCE_DIR}/zip")
+else(USE_BUILTIN_MINIZIP)
+  find_package(PkgConfig)
+  pkg_check_modules(MINIZIP minizip)
+  if(MINIZIP_FOUND)
+    MESSAGE(STATUS "MINIZIP library found at: ${MINIZIP_LIBRARIES}")
+    MESSAGE(STATUS "MINIZIP includes found at: ${MINIZIP_INCLUDE_DIRS}")
+  else()
+    MESSAGE(FATAL_ERROR "MINIZIP not found on your system, try to get this installed. (for example 'sudo apt-get install libminizip-dev')")
+  endif(MINIZIP_FOUND)
+endif(USE_BUILTIN_MINIZIP)
+
+target_include_directories(blebox PRIVATE ${MINIZIP_INCLUDE_DIRS})
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  target_compile_options(blebox PRIVATE
+    -Weverything
+    -Werror
+    -Wno-system-headers
+    -Wno-c++98-compat
+    -Wno-c++98-compat-pedantic
+    -Wno-padded
+    -Wno-documentation-unknown-command
+    -Wno-global-constructors
+
+    # warnings from Domoticz sources:
+    -Wno-zero-as-null-pointer-constant
+    -Wno-extra-semi
+    -Wno-extra-semi-stmt
+    -Wno-reserved-id-macro
+    -Wno-weak-vtables
+    -Wno-double-promotion
+    -Wno-float-conversion
+    -Wno-exit-time-destructors
+
+    # cereal
+    -Wno-sign-conversion
+    )
+endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_compile_options(blebox PRIVATE
+    -Wall
+    -Wextra
+    -Wno-unused-parameter
+    -Wno-missing-field-initializers
+    #-Wno-parentheses
+    )
+endif()
+
+# use cxx-17 for more help from clang-tools
+target_compile_features(blebox PRIVATE cxx_std_17)
```

Comment:
Works on gcc-6 on travis, which - along with cmake - is almost guaranteed to work on any other system. (You can assume that all modern C++ compilers have decent support of c++-17).

Also, the "target_compile_features" is a lot less useful than it sounds. Basically, it's best to support c++-17, unless there's a really, really strong reason to support only c++-14.

That's because there are almost no migration issues/work. (Mostly just the need to upgrade project dependencies from very old pre-c++-11 versions).

There are pretty much only benefits and almost no incompatibilities going from c++-11 to c++17 for this codebase. Besides, the unit/integration tests cover the important parts of MainWorker and SQLHelper, so there's little danger beyond the danger already lurking in any changes in the core codebase. There's also a chance that changes in semantics and strictness could expose existing bugs in Domoticz that no one has found yet.